packages/salesforcedx-utils-vscode/src/cli/diffResultParser.ts

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright (c) 2019, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import { extractJsonObject } from '../helpers';
+
+export interface DiffSuccessResponse {
+  status: number;
+  result: {
+    remote: string;
+    local: string;
+    fileName: string;
+  };
+}
+
+export interface DiffErrorResponse {
+  commandName: string;
+  exitCode: number;
+  message: string;
+  name: string;
+  stack: string;
+  status: number;
+  warnings: any[];
+}
+
+export class DiffResultParser {
+  private response: any;
+
+  constructor(stdout: string) {
+    try {
+      this.response = extractJsonObject(stdout);
+    } catch (e) {
+      const err = new Error('Error parsing diff result');
+      err.name = 'DiffResultParserFail';
+      throw err;
```

Comment:
Do we want to throw an error that warns them if they don't have a more recent enough version of the CLI that bundles the diff plugin?

Reply:
Yes but maybe later, initially we will release it as an additional plugin that needs to be installed on top of the cli.